docs/static/persistent-queues.asciidoc

```diff
@@ -128,24 +128,41 @@ events.
 [[durability-persistent-queues]]
 ==== Controlling Durability
 
+// I'm not sure about this particular description of durability. Might need some help.
+Durability is a property of storage writes that indicate the availability of
+that data after it was written.
+
 When the persistent queue feature is enabled, Logstash will store events on
-disk. The persistent queue exposes the trade-off between performance and
-durability by providing the following configuration options:
+disk. Logstash commits to disk in a mechanism we call checkpointing.
+
+To discuss durability, we need to introduce a few details about how the persistent queue is implemented.
+
+First, the queue itself is a set of pages. There are two kinds of pages: head page and tail page. The head page is where new events are written. When this head page is of a certain size (see `queue.page_capacity), the becomes a tail page, and a new head page is created. Tail pages are immutable, and the head page is append-only.. 
+
+Second, the queue tracks details about the queue (pages, acknowledgements, etc) in a separate file called a checkpoint file.
+
+When recording a checkpoint, Logstash will:
 
-* `queue.checkpoint.writes`: The number of writes to the queue to trigger an
-fsync to disk. This configuration controls the durability from the producer
-side. Keep in mind that a disk flush is a relatively heavy operation that will
-affect throughput if performed after every write. For instance, if you want to
-ensure that all messages in Logstash's queue are durable, you can set
-`queue.checkpoint.writes: 1`. However, this setting can severely impact
-performance.
+* Call fsync on the head page.
+* Atomically write to disk the current state of the queue.
 
-* `queue.checkpoint.acks`: The number of ACKs to the queue to trigger an fsync to disk. This configuration controls the durability from the consumer side.
+The following settings are available to let you tune durability:
 
-The process of checkpointing is atomic, which means any update to the file is
-saved if successful.
+* `queue.checkpoint.writes`: Logstash will checkpoint after this many writes into the queue. At this time, one event counts as one write, but this may change in future releases.
+* `queue.checkpoint.acks`: Logstah will checkpoint after this many events are acknowledged. This configuration controls the durability at the processing (filter + output)
+part of Logstash.
+
+Disk writes are have a resource cost. Tuning the above values higher or lower will trade durability for performance. For instance, if you want to the strongest durability for all input events, you can set `queue.checkpoint.writes: 1`.
```

Comment:
You're right, I should contract it to `are've` an obvious contraction of "are" and "have"..

haha ;)